src/Storage/Storage/File/Cmdlet/CloseAzureStorageFileHandle.cs

```diff
@@ -0,0 +1,294 @@
+            Position = 1,
+            Mandatory = false,
+            ParameterSetName = DirectoryCloseAllParameterSetName,
+            HelpMessage = "Path to an existing file/directory.")]
+        public string Path { get; set; }
+
+        [Parameter(Mandatory = true, ParameterSetName = ShareNameCloseSingleParameterSetName, ValueFromPipeline = true, HelpMessage = "The File Handle to close.")]
+        [Parameter(Mandatory = true, ParameterSetName = ShareCloseSingleParameterSetName, ValueFromPipeline = true, HelpMessage = "The File Handle to close.")]
+        [ValidateNotNull]
+        public PSFileHandle FileHandle { get; set; }
+
+        [Parameter(Mandatory = false, ParameterSetName = ShareNameCloseAllParameterSetName, HelpMessage = "Closed handles Recursively. Only works on File Directory.")]
+        [Parameter(Mandatory = false, ParameterSetName = ShareCloseAllParameterSetName, HelpMessage = "Closed handles Recursively. Only works on File Directory.")]
+        [Parameter(Mandatory = false, ParameterSetName = DirectoryCloseAllParameterSetName, HelpMessage = "Closed handles Recursively. Only works on File Directory.")]
+        public SwitchParameter Recursive { get; set; }
+
+        [Parameter(Mandatory = true, ParameterSetName = ShareNameCloseAllParameterSetName, HelpMessage = "Force close all File handles.")]
+        [Parameter(Mandatory = true, ParameterSetName = ShareCloseAllParameterSetName, HelpMessage = "Force close all File handles.")]
+        [Parameter(Mandatory = true, ParameterSetName = DirectoryCloseAllParameterSetName, HelpMessage = "Force close all File handles.")]
+        [Parameter(Mandatory = true, ParameterSetName = FileCloseAllParameterSetName, HelpMessage = "Force close all File handles.")]
+        public SwitchParameter CloseAll { get; set; }
+
+        [Parameter(
+            ValueFromPipeline = true,
+            ValueFromPipelineByPropertyName = true,
+            ParameterSetName = ShareNameCloseSingleParameterSetName,
+            HelpMessage = "Azure Storage Context Object")]
+        [Parameter(
+            ValueFromPipeline = true,
+            ValueFromPipelineByPropertyName = true,
+            ParameterSetName = ShareNameCloseAllParameterSetName,
+            HelpMessage = "Azure Storage Context Object")]
+        public override IStorageContext Context { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = "Return whether the specified blob is successfully removed")]
+        public SwitchParameter PassThru { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = "Run cmdlet in the background")]
+        public virtual SwitchParameter AsJob { get; set; }
+
+        public override void ExecuteCmdlet()
+        {
+            if (this.ShouldProcess(string.Format("Close File Handles for File or FileDirectory on Path: {0}", Path), "This operation will force the provided file handle(s) closed, which may cause data loss or corruption for active applications/users.", null))
```

Comment:
suggestion: It would be much more useful if you used the SHouldProcess to guard the individual file closings - this way a user would get an entire listing of the file closings that would take place.

This is just a suggestion, it is not required

Reply:
@markcowl  
I don't quite understand what you means, but for this cmdlets, if input a File Dir to close file handle, we will not know the name of each files that will handles, server will list the handles in the dir, and close them.